### PR TITLE
adds dns validation step after dhclient restart

### PR DESCRIPTION
**Pull request by andkononykhin2**
- adds limited awaiting for DNS to set up after dhclient restart

### CODE REVIEW

File: srv/components/system/network/prepare.sls [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 {% if nwmanager_is_running %}
 
+    {% set hostname_short = salt['cmd.run']('hostname -s') %}
```
Comment (ypise) [inline srv/components/system/network/prepare.sls:22]: Should be `cmd.shell` if we are capturing the value.

File: srv/components/system/network/files/dns_check.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#
+
+
+set -eu
+
+domain="${1:-$(hostname)}"
+
+echo "Checking DNS resolution for domain '$domain'"
+
+ns_servers=$(dig +short NS "$domain" | paste -sd " " -)
```
Comment (ypise) [inline srv/components/system/network/files/dns_check.sh:26]: Should we check if dig is installed?

File: srv/components/system/network/prepare.sls (hunk 1 above)
Comment (andkononykhin2) [reply srv/components/system/network/prepare.sls:22]: cmd.run for worked the same

File: srv/components/system/network/files/dns_check.sh (hunk 2 above)
Comment (andkononykhin2) [reply srv/components/system/network/files/dns_check.sh:26]: bind-utils provides it and we do install them [here](https://github.com/Seagate/cortx-prvsnr/blob/72d5c1e2a8375de6f8548791a4123770ac17bba8/srv/components/system/install.sls#L23)

Comment (andkononykhin2) [reply srv/components/system/network/files/dns_check.sh]: resolve that as `getent` now is used instead

**Review by ypise — APPROVED** [review]

**Pull request merged**